document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/service/impl/DocumentGeneratorServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+
+    private DocumentInfoService documentInfoService;
+
+    private EnvironmentReader environmentReader;
+
+    private RenderDocumentRequestHandler requestHandler;
+
+    private static final String DOCUMENT_RENDER_SERVICE_HOST_ENV_VAR = "DOCUMENT_RENDER_SERVICE_HOST";
+
+    private static final String CONTEXT_PATH = "/document-render/store?is_public=true";
+
+    private static final Logger LOG = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    @Autowired
+    public DocumentGeneratorServiceImpl(DocumentInfoService documentInfoService, EnvironmentReader environmentReader,
+                                        RenderDocumentRequestHandler requestHandler) {
+
+        this.documentInfoService = documentInfoService;
+        this.environmentReader = environmentReader;
+        this.requestHandler = requestHandler;
+    }
+
+    @Override
+    public DocumentResponse generate(DocumentRequest documentRequest) {
+
+        DocumentInfo documentInfo = null;
+        DocumentResponse response = null;
+        RenderDocumentResponse renderResponse;
+
+        //TODO addition of get doc gen type from URL to be added in SFA 580
+
+        //TODO currently no impl present, being completed in SFA 567
+        documentInfo = documentInfoService.getDocumentInfo();
+
+        if (documentInfo != null) {
+            renderResponse = renderSubmittedDocumentData(documentRequest, documentInfo);
+            response = setDocumentResponse(renderResponse, documentInfo);
+        } else {
+            //TODO currently no impl present so errors not confirmed, being completed in SFA 567
+           Exception e  = new Exception("No data returned from documentInfoService");
+           LOG.error(e);
+        }
+
+        return response;
+    }
+
+    /**
+     * Send data to Render Service and generate document
+     *
+     * @param documentRequest
+     * @param documentInfo
+     * @return A populated RenderDocumentResponse model or Null
+     */
+    private RenderDocumentResponse renderSubmittedDocumentData(DocumentRequest documentRequest,
+                                                               DocumentInfo documentInfo) {
+
+        String host = environmentReader.getMandatoryString(DOCUMENT_RENDER_SERVICE_HOST_ENV_VAR);
+        String url = host + CONTEXT_PATH;
+
+        RenderDocumentRequest requestData = new RenderDocumentRequest();
+        requestData.setAssetId(documentInfo.getAssetId());
+        requestData.setContentType(documentRequest.getContentType());
+        requestData.setData(documentInfo.getData());
+        requestData.setDocumentType(documentRequest.getDocumentType());
+        requestData.setTemplateName(documentInfo.getTemplateName());
+        requestData.setLocation(documentInfo.getLocation());
+
+        try {
+            return requestHandler.sendDataToDocumentRenderService(url, requestData);
+        } catch (Exception e) {
+            LOG.error(e);
+        }
+
+        return null;
```

Comment:
Should we be returning null here? In the `catch` case I would assume that we'd cascade the exception to the controller

Reply:
yes we should, need to still populate a response further on, but this will return null as it has failed to render document, so no location or size present.

Reply:
We agreed in the exception handling discussion that we would throw exceptions up to the controller. The exception in the catch block should be logged, wrapped in an exception relevant to this layer and thrown up to the controller to be converted into the correct HTTP status code.